### ISSUE TITLE
`explore`: panic when giving `where size > 0b` to `try` 

### ISSUE THREAD

**Issue by webbedspace** [nushell/nushell#7323]
### Describe the bug

Here is the backtrace:
```
 ╔════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════╗ 
 ║ ls | where size > 0bthread 'main' panicked at 'internal error: missing block', crates\nu-protocol\src\engine\engine_state.rs:724:14                                                                                                                                        ║ 
stack backtrace:══════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════╝ 
   0:     0x7ff7dcf232d2 - std::backtrace_rs::backtrace::dbghelp::trace───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐ 
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120/library\std\src\..\..\backtrace\src\backtrace\dbghelp.rs:98                                                                                                                                 │ 
   1:     0x7ff7dcf232d2 - std::backtrace_rs::backtrace::trace_unsynchronized                                                                                                                                                                                                 │ 
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120/library\std\src\..\..\backtrace\src\backtrace\mod.rs:66                                                                                                                                     │ 
   2:     0x7ff7dcf232d2 - std::sys_common::backtrace::_print_fmt                                                                                                                                                                                                             │ 
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120/library\std\src\sys_common\backtrace.rs:66                                                                                                                                                  │ 
   3:     0x7ff7dcf232d2 - std::sys_common::backtrace::_print::impl$0::fmt                                                                                                                                                                                                    │ 
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120/library\std\src\sys_common\backtrace.rs:45                                                                                                                                                  │ 
   4:     0x7ff7dcf4a27b - core::fmt::write                                                                                                                                                                                                                                   │ 
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120/library\core\src\fmt\mod.rs:1202                                                                                                                                                            │ 
   5:     0x7ff7dcf1973a - std::io::Write::write_fmt<std::sys::windows::stdio::Stderr>                                                                                                                                                                                        │ 
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120/library\std\src\io\mod.rs:1679                                                                                                                                                              │ 
   6:     0x7ff7dcf26134 - std::sys_common::backtrace::_print                                                                                                                                                                                                                 │ 
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120/library\std\src\sys_common\backtrace.rs:48                                                                                                                                                  │ 
   7:     0x7ff7dcf26134 - std::sys_common::backtrace::print                                                                                                                                                                                                                  │ 
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120/library\std\src\sys_common\backtrace.rs:35                                                                                                                                                  │ 
   8:     0x7ff7dcf26134 - std::panicking::default_hook::closure$1                                                                                                                                                                                                            │ 
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120/library\std\src\panicking.rs:295                                                                                                                                                            │ 
   9:     0x7ff7dcf25d6a - std::panicking::default_hook                                                                                                                                                                                                                       │ 
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120/library\std\src\panicking.rs:314                                                                                                                                                            │ 
  10:     0x7ff7daa0b0a0 - alloc::boxed::impl$46::call<tuple$<ref$<core::panic::panic_info::PanicInfo> >,dyn$<core::ops::function::Fn<tuple$<ref$<core::panic::panic_info::PanicInfo> >,assoc$<Output,tuple$<> > >,core::marker::Send,core::marker::Sync>,alloc::alloc::Global> 
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120\library\alloc\src\boxed.rs:1954                                                                                                                                                             │ 
  11:     0x7ff7da9fec49 - nu::main::closure$0                                                                                                                                                                                                                                │ 
                               at C:\Coding\nushell\src\main.rs:142                                                                                                                                                                                                           │ 
  12:     0x7ff7dcf26aa2 - std::panicking::rust_panic_with_hook                                                                                                                                                                                                               │ 
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120/library\std\src\panicking.rs:702                                                                                                                                                            │ 
  13:     0x7ff7dcf2682d - std::panicking::begin_panic_handler::closure$0─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘ 
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120/library\std\src\panicking.rs:588
  14:     0x7ff7dcf23fdf - std::sys_common::backtrace::__rust_end_short_backtrace<std::panicking::begin_panic_handler::closure_env$0,never$>
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120/library\std\src\sys_common\backtrace.rs:138
  15:     0x7ff7dcf264f0 - std::panicking::begin_panic_handler
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120/library\std\src\panicking.rs:584
  16:     0x7ff7dd0b4cc5 - core::panicking::panic_fmt
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120/library\core\src\panicking.rs:142
  17:     0x7ff7dcf47c70 - core::panicking::panic_display<str>
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120/library\core\src\panicking.rs:72
  18:     0x7ff7dcf47c1b - core::panicking::panic_str
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120/library\core\src\panicking.rs:56
  19:     0x7ff7dd0b4b19 - core::option::expect_failed
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120/library\core\src\option.rs:1880
  20:     0x7ff7dca1c9cc - enum2$<core::option::Option<ref$<nu_protocol::ast::block::Block> > >::expect<ref$<nu_protocol::ast::block::Block> >
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120\library\core\src\option.rs:738
  21:     0x7ff7dc9c3f0d - nu_protocol::engine::engine_state::EngineState::get_block
                               at C:\Coding\nushell\crates\nu-protocol\src\engine\engine_state.rs:722
  22:     0x7ff7dc88b78f - nu_engine::eval::eval_expression
                               at C:\Coding\nushell\crates\nu-engine\src\eval.rs:538
  23:     0x7ff7dab21064 - nu_engine::call_ext::impl$0::opt<nu_protocol::engine::capture_block::Closure>
                               at C:\Coding\nushell\crates\nu-engine\src\call_ext.rs:77
  24:     0x7ff7db02f3e3 - nu_command::filters::where_::impl$0::run
                               at C:\Coding\nushell\crates\nu-command\src\filters\where_.rs:212
  25:     0x7ff7dc885b84 - nu_engine::eval::eval_call
                               at C:\Coding\nushell\crates\nu-engine\src\eval.rs:177
  26:     0x7ff7dc8904be - nu_engine::eval::eval_expression_with_input
                               at C:\Coding\nushell\crates\nu-engine\src\eval.rs:696
  27:     0x7ff7dc8911ae - nu_engine::eval::eval_element_with_input
                               at C:\Coding\nushell\crates\nu-engine\src\eval.rs:826
  28:     0x7ff7dc892ae5 - nu_engine::eval::eval_block
                               at C:\Coding\nushell\crates\nu-engine\src\eval.rs:1002
  29:     0x7ff7db405093 - nu_explore::nu_common::command::eval_source2
                               at C:\Coding\nushell\crates\nu-explore\src\nu_common\command.rs:53
  30:     0x7ff7db404bdc - nu_explore::nu_common::command::run_nu_command
                               at C:\Coding\nushell\crates\nu-explore\src\nu_common\command.rs:15
  31:     0x7ff7db3f36c8 - nu_explore::views::interative::impl$1::handle_input
                               at C:\Coding\nushell\crates\nu-explore\src\views\interative.rs:185
  32:     0x7ff7db3ed17d - nu_explore::views::impl$2::handle_input
                               at C:\Coding\nushell\crates\nu-explore\src\views\mod.rs:89
  33:     0x7ff7db3bf119 - nu_explore::pager::handle_events<alloc::boxed::Box<dyn$<nu_explore::views::View>,alloc::alloc::Global> >
                               at C:\Coding\nushell\crates\nu-explore\src\pager.rs:462
  34:     0x7ff7db3b07e8 - nu_explore::pager::render_ui
                               at C:\Coding\nushell\crates\nu-explore\src\pager.rs:185
  35:     0x7ff7db3aff37 - nu_explore::pager::run_pager
                               at C:\Coding\nushell\crates\nu-explore\src\pager.rs:111
  36:     0x7ff7db3b3534 - nu_explore::pager::Pager::run
                               at C:\Coding\nushell\crates\nu-explore\src\pager.rs:782
  37:     0x7ff7db3b5d74 - nu_explore::run_pager
                               at C:\Coding\nushell\crates\nu-explore\src\lib.rs:57
  38:     0x7ff7daf99311 - nu_command::viewers::explore::impl$0::run
                               at C:\Coding\nushell\crates\nu-command\src\viewers\explore.rs:78
  39:     0x7ff7dc885b84 - nu_engine::eval::eval_call
                               at C:\Coding\nushell\crates\nu-engine\src\eval.rs:177
  40:     0x7ff7dc8904be - nu_engine::eval::eval_expression_with_input
                               at C:\Coding\nushell\crates\nu-engine\src\eval.rs:696
  41:     0x7ff7dc8911ae - nu_engine::eval::eval_element_with_input
                               at C:\Coding\nushell\crates\nu-engine\src\eval.rs:826
  42:     0x7ff7dc892ae5 - nu_engine::eval::eval_block
                               at C:\Coding\nushell\crates\nu-engine\src\eval.rs:1002
  43:     0x7ff7dbb480c0 - nu_cli::util::eval_source
                               at C:\Coding\nushell\crates\nu-cli\src\util.rs:231
  44:     0x7ff7dbb508cb - nu_cli::repl::evaluate_repl
                               at C:\Coding\nushell\crates\nu-cli\src\repl.rs:429
  45:     0x7ff7daa0e3e5 - nu::main
                               at C:\Coding\nushell\src\main.rs:440
  46:     0x7ff7da9fdbab - core::ops::function::FnOnce::call_once<enum2$<core::result::Result<tuple$<>,miette::eyreish::Report> > (*)(),tuple$<> >
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120\library\core\src\ops\function.rs:248
  47:     0x7ff7daa1581e - std::sys_common::backtrace::__rust_begin_short_backtrace<enum2$<core::result::Result<tuple$<>,miette::eyreish::Report> > (*)(),enum2$<core::result::Result<tuple$<>,miette::eyreish::Report> > >
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120\library\std\src\sys_common\backtrace.rs:122
  48:     0x7ff7daa0c291 - std::rt::lang_start::closure$0<enum2$<core::result::Result<tuple$<>,miette::eyreish::Report> > >
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120\library\std\src\rt.rs:166
  49:     0x7ff7dcf119de - core::ops::function::impls::impl$2::call_once
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120/library\core\src\ops\function.rs:283
  50:     0x7ff7dcf119de - std::panicking::try::do_call
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120/library\std\src\panicking.rs:492
  51:     0x7ff7dcf119de - std::panicking::try
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120/library\std\src\panicking.rs:456
  52:     0x7ff7dcf119de - std::panic::catch_unwind
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120/library\std\src\panic.rs:137
  53:     0x7ff7dcf119de - std::rt::lang_start_internal::closure$2
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120/library\std\src\rt.rs:148
  54:     0x7ff7dcf119de - std::panicking::try::do_call
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120/library\std\src\panicking.rs:492
  55:     0x7ff7dcf119de - std::panicking::try
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120/library\std\src\panicking.rs:456
  56:     0x7ff7dcf119de - std::panic::catch_unwind
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120/library\std\src\panic.rs:137
  57:     0x7ff7dcf119de - std::rt::lang_start_internal
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120/library\std\src\rt.rs:148
  58:     0x7ff7daa0c26a - std::rt::lang_start<enum2$<core::result::Result<tuple$<>,miette::eyreish::Report> > >
                               at /rustc/897e37553bba8b42751c67658967889d11ecd120\library\std\src\rt.rs:165
  59:     0x7ff7daa13edc - main
  60:     0x7ff7dd09b29c - invoke_main
                               at D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:78
  61:     0x7ff7dd09b29c - __scrt_common_main_seh
                               at D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288
  62:     0x7ffd625874b4 - BaseThreadInitThunk
  63:     0x7ffd644826a1 - RtlUserThreadStart
```

### How to reproduce

1. Run `ls | explore`
2. Inside explore, run `:try`
3. Input `where size > 0b`

### Expected behavior

…

### Screenshots

_No response_

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.72.0                                   |
| branch             |                                          |
| commit_hash        | 8fa965118c2e5a0beaa528d07eaf9b33d96779cb |
| build_os           | windows-x86_64                           |
| build_target       | x86_64-pc-windows-msvc                   |
| rust_version       | rustc 1.65.0 (897e37553 2022-11-02)      |
| rust_channel       | stable-x86_64-pc-windows-msvc            |
| cargo_version      | cargo 1.65.0 (4bc8f24d3 2022-10-20)      |
| pkg_version        | 0.72.0                                   |
| build_time         | 2022-11-29 18:11:35 +00:00               |
| build_rust_channel | release                                  |
| features           | database, default, trash, which, zip     |
| installed_plugins  |                                          |

### Additional context

_No response_

**Comment by zhiburt** [comment]
Sorry....

Must be fixed with #7334 could you verify?

**Comment by fdncred** [comment]
fix works for me. thanks @zhiburt!!
